Add whois: <server> type referral regex

martijn opened the pull request:
For certain queries, ARIN whois server skips the `refer:` reponse and only returns a `whois:` response, i.e.:

```
% IANA WHOIS server
% for more information on IANA, visit http://www.iana.org
% This query returned 1 object

inet6num:     xxxx::/23
organisation: RIPE NCC
status:       ALLOCATED

whois:        whois.ripe.net

changed:      2004-06-11
source:       IANA
```

This changes ensures that gets picked up.

jsakamoto commented:
@martijn Could you tell me a real example of the command using `whois` to show the same output as above? I need to reproduce it on my hand and add a new test for that.

martijn commented:
Sorry for the late reply, I missed the notification.

`whois.iana.org` does this for IPv6 addresses, like `2600::` for example. macOS `whois` follows this type of referral:

```
$ whois 2600::
% IANA WHOIS server
% for more information on IANA, visit http://www.iana.org
% This query returned 1 object

inet6num:     2600:0:0:0:0:0:0:0/12
organisation: ARIN
status:       ALLOCATED

whois:        whois.arin.net

changed:      2006-10-03
source:       IANA

# whois.arin.net

NetRange:       2600:: - 2600:F:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF
CIDR:           2600::/28
NetName:        SPRINT-V6IP
NetHandle:      NET6-2600-2
Parent:         NET6-2600 (NET6-2600-1)
NetType:        Direct Allocation
OriginAS:
Organization:   Sprint (SPRN)
RegDate:        2006-12-21
Updated:        2023-01-04
Ref:            https://rdap.arin.net/registry/ip/2600::
```

jsakamoto commented:
@martijn I merged your pull request, and I also added a test case to verify your code changes work fine.
After that, I published the new version of "WhoisClient.NET" today.

- https://www.nuget.org/packages/WhoisClient.NET/5.0.0

Thank you for your contributions!
